### ISSUE TITLE
Division object is projected to next frame

### ISSUE THREAD

**Issue by constantinpape** [computational-cell-analytics/micro-sam#113]
Reported by @Sagnik700 in #64 

Testing micro-sam with [0.1.2.post1](https://github.com/computational-cell-analytics/micro-sam/releases/tag/0.1.2.post1)  version and with [MiaPaCa200x200](https://owncloud.gwdg.de/index.php/s/mTyqEdgg9xmVEkv) dataset, daughter cell segmentation of a frame was visible in the previous frame where the parent cell was annotated as `division` in `track_state`. Screenshot 1(Frame 22) shows the **parent cell segmentation in brown** and the **daughter cell segmentation in blue**. The blue segmentation should not be visible in Screenshot 1 since it is appearing in Screenshot 2(Frame 23) onwards.

Steps to replicate the issue:
1. Segment the exact cell which is highlighted as `division` as `track_state` in frame 22
2. Segment the daughter cell as highlighted as `track 2` as `track_state` in frame 23 and click on `Track Object`
3. Come back to frame 22 to notice the issue

![Screenshot 1](https://github.com/computational-cell-analytics/micro-sam/assets/23014134/617c6a54-5768-49a7-a01b-e35b59bf9450)
![Screenshot 2](https://github.com/computational-cell-analytics/micro-sam/assets/23014134/e0d14781-15a6-4bd5-a3bd-cbc50011ee7f)

_Originally posted by @Sagnik700 in https://github.com/computational-cell-analytics/micro-sam/issues/64#issuecomment-1660880123_

**Comment by constantinpape** [comment]
Hi @Sagnik700 ,
I created a separate issue for this. It should be fixed by #114 now, but I haven't checked if it's really fixed.
Can you pull the master and confirm that it's fixed?
Thanks!

**Comment by Sagnik700** [comment]
Yes @constantinpape I just checked now and the issue seems to be fixed!!

**Comment by Sagnik700** [comment]
Reopening the issue since the issue reappeared while checking with [117](https://github.com/computational-cell-analytics/micro-sam/pull/117)
